### DISCUSSION TITLE
Tooltips on Dropdown Menus

### DISCUSSION THREAD

**Discussion by mathieu-fournier** [iTwin/iTwinUI-react#296] [General]
### Tooltips on Dropdown Menus

For anyone wondering how to add a Tooltip on a DropdownButton or on the button to open a DropdownMenu. 
Wrapping them in a div makes it work. 

See pictures below for examples : 
![image](https://user-images.githubusercontent.com/82905333/130993566-f24ca5f2-0b16-4358-b077-ad3de4a26fa6.png)
![image](https://user-images.githubusercontent.com/82905333/130993517-4155e5e6-bf6a-4752-afc9-8e5b8b83c031.png)

### Note: 
It doesn't work without the div because Tooltip and DropdownButton are function components, which cannot be used as Refs by their parents.
![image](https://user-images.githubusercontent.com/82905333/130994327-40c1fee9-81cb-4e71-b43c-62a8f4fc0adc.png)

### Also:
Is this "div wrapper" workflow common in React or is it an anti-pattern I found ?

**Comment by mayank99** [comment]
Hi @mathieu-fournier 👋 Thanks for creating this discussion.

The "div wrapper workflow" is actually documented in our [FAQ](https://github.com/iTwin/iTwinUI-react/wiki/FAQ#popover-tooltips-dropdowns-etc).
![image](https://user-images.githubusercontent.com/9084735/131008251-c89656fa-37da-40aa-abbf-ff5946c99cce.png)

For your two cases,

1. The `DropdownButton` case may actually be considered a bug, because unlike our other buttons, we do not forward its ref. We'll see if this can be fixed in the next release.

2. The `DropdownMenu` case is more interesting. Since we cannot always be sure of the trigger component (and therefore cannot forward its ref), using a wrapper is expected. However, I'm curious about your example setup with a `div` inside children. Does the dropdown menu open and the tooltip also functions correctly?

**Reply by mathieu-fournier** [reply]
Thanks for the response :).
Yes, both case mentioned functions correctly for the DropdownMenu.

**Reply by mayank99** [reply]
Update: you don't need a wrapper div for `DropdownButton` anymore (fixed in v1.17.0). 🎉

And for the `DropdownMenu` case, you could use `reference` to avoid the extra div:
![](https://user-images.githubusercontent.com/9084735/131517111-23db83f2-4dde-48a4-9dba-96bb9c7f2f63.png)
